ssl/quic/quic_channel.c

```diff
@@ -3809,7 +3810,7 @@ int ossl_quic_bind_channel(QUIC_CHANNEL *ch, const BIO_ADDR *peer,
                            const QUIC_CONN_ID *peer_dcid,
                            const QUIC_CONN_ID *peer_odcid)
 {
-    if (peer_dcid == NULL)
+    if (peer_dcid == NULL || peer_scid == NULL)
```

Comment:
the `peer_scid` is currently local variable defined in `port_default_packet_handler()` which reaches here, so peer_scid should never ever be NULL. at least for now. So in my opinion we should just drop this change at least for now.

Leaving code as will allow us to find issues possibly introduced by refactoring done in the future more quickly.

Reply:
another option is to add line `OPENSSL_assert(peer_scid != NULL);`

Reply:
I agree to have an assert line as @Sashan suggested.

> another option is to add line `OPENSSL_assert(peer_scid != NULL);`